Move global arrays to function scope

saschazar21 opened the pull request:
As suggested by [@ddprrt](https://github.com/ddprrt), linting errors are shown file by file, if errors & allchecks arrays are moved to function scope.
parseChunk gets enhanced by the errors array parameter, possible linting errors therefore get pushed by reference.

saschazar21 committed:
Move global arrays to function scope

As suggested by @ddprrt, linting errors are shown file by file, if errors & allchecks arrays are moved to function scope.
parseChunk gets enhanced by the errors array parameter, possible linting errors therefore get pushed by reference.

ddprrt commented:
Original problem is that with lots of parallel request the global errors object is being used by multiple requests at the same time.

tomheller commented:
I will need to check if this breaks the atom-liquid-linter... give me a bit of time

saschazar21 committed:
Added errors parameter to return function

The errors parameter has to be included in the recursive function call as well.

tomheller commented:
@saschazar21 looks good, can you bump up the package.json version up to 0.0.4?

saschazar21 committed:
bump to 0.0.4
